zerver/lib/export.py

```diff
@@ -976,28 +984,71 @@ def get_ids(records: List[Record]) -> Set[int]:
         response['zerver_userprofile_mirrordummy'] +
         response['zerver_userprofile_crossrealm'])
 
-    if public_only:
+    consented_user_ids = set()  # type: Set[int]
+    if consent_message_id is not None:
+        consented_user_ids = get_consented_user_ids(consent_message_id)
+
+    if len(consented_user_ids) != 0:
```

Comment:
I think this should be `if consent_user_id is not None`; we want to use this code path even if 0 users consent, rather than falling back to a full export, right?